pkg/app/carbonapi/http_handlers.go

```diff
@@ -338,6 +339,23 @@ func (app *App) renderHandler(w http.ResponseWriter, r *http.Request, lg *zap.Lo
 	}
 }
 
+func clustersFromMetricMap(metricMap map[parser.MetricRequest][]*types.MetricData) []string {
```

Comment:
That looks like quite heavy function, and we're calling it for every metric. Maybe we can cache/memoize result of it?

Reply:
Ok, discussed offline, maybe it's excessive.